.github/workflows/spread-tests.yaml

```diff
@@ -101,32 +131,39 @@ jobs:
     - name: Setup run tests variable
       if: "!contains(github.event.pull_request.labels.*.name, 'Skip spread')"
       run: |
+          get_new_tests() {
+            # The tests are just filtered when the change is a PR
+            # When 'Run Nested' label is added in a PR, all the nested tests have to be executed
+            TESTS_TO_RUN=""
+            if [ -z "${{ github.event.number }}" ] || [ "$RUN_NESTED" = 'true' ]; then
+                for TESTS in ${{ inputs.tasks }}; do
+                    TESTS_TO_RUN="$TESTS_TO_RUN ${{ inputs.backend }}$1:$TESTS"
+                done
+            else
+                NEW_TESTS="$(./tests/lib/external/snapd-testing-tools/utils/spread-filter -r ./tests/lib/spread/rules/${{ inputs.rules }}.yaml -p "${{ inputs.backend }}$1" $2)"
+                TESTS_TO_RUN="$NEW_TESTS"
+            fi
+            echo "$TESTS_TO_RUN"
+          }
+
+          CHANGES_PARAM=""
+          for CHANGE in $CHANGED_FILES; do
+              CHANGES_PARAM="$CHANGES_PARAM -c $CHANGE"
+          done
           RUN_TESTS=""
-          SUGGESTED_TESTS=""
           # Save previous failed test results in FAILED_TESTS env var
           if [ -n "$FAILED_TESTS" ]; then
               RUN_TESTS="$FAILED_TESTS"
-          else
-              CHANGES_PARAM=""
-              for CHANGE in $CHANGED_FILES; do
-                  CHANGES_PARAM="$CHANGES_PARAM -c $CHANGE"
-              done
+          elif [ "${{ inputs.systems }}" == "ALL" ]; then
+              RUN_TESTS=$(get_new_tests "" "$CHANGES_PARAM")
+          elif [ "${{ inputs.systems }}" != "NONE" ]; then
               for SYSTEM in ${{ inputs.systems }}; do
-                  # Configure parameters to run tests based on current changes
-                  # The tests are just filtered when the change is a PR
-                  # When 'Run Nested' label is added in a PR, all the nested tests have to be executed
-                  if [ -z "${{ github.event.number }}" ] || [ "$RUN_NESTED" = 'true' ]; then
-                      for TESTS in ${{ inputs.tasks }}; do
-                          RUN_TESTS="$RUN_TESTS ${{ inputs.backend }}:$SYSTEM:$TESTS"
-                      done
-                  else
-                      NEW_TESTS="$(./tests/lib/external/snapd-testing-tools/utils/spread-filter -r ./tests/lib/spread/rules/${{ inputs.rules }}.yaml -p "${{ inputs.backend }}:$SYSTEM" $CHANGES_PARAM)"
-                      if [ -z "$RUN_TESTS" ]; then
-                          RUN_TESTS="$NEW_TESTS"
-                      else
-                          RUN_TESTS="$RUN_TESTS $NEW_TESTS"
-                      fi
-                  fi
+                NEW_TESTS="$(get_new_tests ":$SYSTEM" "$CHANGES_PARAM")"
```

Comment:
You could call here 
NEW_TESTS="$(get_new_tests "${{ inputs.backend }}:$SYSTEM" "$CHANGES_PARAM")"